cli_tools/common/utils/logging/service/logger.go

```diff
@@ -381,6 +386,34 @@ func (l *Logger) constructLogRequest(logExtension *ComputeImageToolsLogExtension
 	return reqStr, err
 }
 
+// safeRunner supports running a function that may panic. If there's a panic, execution is
+// recovered and the panic's details are captured.
+type safeRunner struct {
+	inner    func() (Loggable, error)
+	loggable Loggable
+	err      error
+}
+
+// runInner executes the function `inner`, and captures the results in the fields loggable and err.
+// If a panic occurs during the execution of `inner`, it is trapped, and the panic's contents are
+// used to create loggable and err.
+func (p *safeRunner) runInner() {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Printf("Fatal error: %v", err)
+			p.loggable = literalLoggable{
+				traceLogs: []string{
+					fmt.Sprintf("Captured panic: %v", err),
+					"stacktrace from panic: \n" + string(debug.Stack()),
+				},
+			}
+			p.err = errors.New("A fatal error has occurred. " +
+				"Please submit an issue at https://github.com/GoogleCloudPlatform/compute-image-tools/issues")
```

Comment:
Does this overwrite what p.inner() returned as an error? Does that one get lost? Or is that one nil in case of panic?

Reply:
From my reading of [1]:
1. If inner panics, it doesn't return, so `p.err` won't be written at 414.
 1. If inner didn't panic, then `recover()` returns nil, so this block won't get executed.